khal/khalendar/event.py

```diff
@@ -881,13 +881,14 @@ def create_timezone(tz, first_date=None, last_date=None):
     easy solution, we'd really need to ship another version of the OLSON DB.
 
     """
+
     if isinstance(tz, pytz.tzinfo.StaticTzInfo):
         return _create_timezone_static(tz)
 
     # TODO last_date = None, recurring to infinity
 
     first_date = dt.datetime.today() if not first_date else to_naive_utc(first_date)
-    last_date = dt.datetime.today() if not last_date else to_naive_utc(last_date)
+    last_date = first_date + dt.timedelta(days=1) if not last_date else to_naive_utc(last_date)
```

Comment:
I don't quite follow why/how this fixes the issue at hand.

Reply:
The main difference here is if `first_date` is set, but not `last_date`. The old version would use `today()` which can be earlier than `first_date`.